### ISSUE TITLE
Add case conversion to ImplicitDefaultLocale

### ISSUE THREAD

**Issue by chao2zhang** [arturbosch/detekt#2515]
##  Behavior of the rule
ImplicitDefaultLocale should not only consider `String.format()`, but also case conversions like `String.toUpperCase()` and `String.toLowerCase()`. 

For example,
```
val s: String = getString()
val t: String? = getString()
s.toUpperCase() // Should be reported
t?.toLowerCase() // Should be reported
s.toLowerCase(Locale.US) // Should not be reported
t?.toUpperCase(Locale.US) // Should not be reported.
```
should be reported.

## Context
The reason for case conversion is the same as formatting function: Default locale may result in unexpected result, so we should recommend developers to pass in Locale.US.

Android has a [lint check](https://cs.android.com/android/platform/superproject/+/master:tools/base/lint/libs/lint-checks/src/main/java/com/android/tools/lint/checks/LocaleDetector.java?q=LocaleDetector) that checks both case conversion and formatting. Note that it is running the check on the loaded JVM class, which should cover all edge cases like `s.let(it::toUpperCase)`. However, I believe most of the violations will be in its vanilla format which can be caught by detekt, which is also satisfactory to me.
